### ISSUE TITLE
CodeSnippet copy button resets tab index/focus

### ISSUE THREAD

**Issue by Zoe-Gathercole1** [carbon-design-system/carbon#6163]
## Environment

> Operating system
MacOS

> Browser
Chrome/Firefox

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?
10

> What did you expect to happen?
When pressing enter on the copy button I assumed the focus / tab index would return to a logical position.

> What happened instead?
The focus was reset to the start / top of the page

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Tab to code snippet copy button
2. Press enter
3. Press tab again

## Additional information

The DOM changes when you invoke the onClick of the code snippet as the copy button introduces the assistive text so it's understandable why the focus moves back to the start of the page but this is not logical and unhelpful for those using the keyboard to navigate as it takes a while to get back to where they were. 

There is no option / prop to determine where the focus is put back to after the copy action is invoked so we cannot change it. Is this behaviour to be expected or can it be controlled another way? 

Related issue: https://github.ibm.com/mhub/qp-planning/issues/5510

**Comment by emyarod** [comment]
I am not able to reproduce this behavior in our storybook. 
can you create a [reduced test case in Code Sandbox](https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox) for easier testing and debugging?

**Comment by matthew-chirgwin** [comment]
Hi @emyarod - I work with @Zoe-Gathercole1 . I just tried to use the code sandbox to reproduce the issue we are seeing I am getting errors back (with the snippet with the Button as is):

```
TypeError
mixinfn is not a function
mixin/<
https://uymgo.csb.app/node_modules/carbon-components/es/globals/js/misc/mixin.js:41:12
mixin
https://uymgo.csb.app/node_modules/carbon-components/es/globals/js/misc/mixin.js:40:28
evaluate
https://uymgo.csb.app/node_modules/carbon-components/es/components/floating-menu/floating-menu.js:343:4
bo
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:197349
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:214199
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:237041
t
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:213999
evaluate
https://uymgo.csb.app/node_modules/carbon-components/es/components/overflow-menu/overflow-menu.js:117:26
bo
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:197349
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:214199
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:237041
t
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:213999
evaluate
https://uymgo.csb.app/node_modules/carbon-components/es/globals/js/components.js:12:26
bo
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:197349
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:214199
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:237041
t
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:213999
evaluate
https://uymgo.csb.app/node_modules/carbon-components/es/index.js:4:24
bo
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:197349
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:214199
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:237041
t
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:213999
evaluate
https://uymgo.csb.app/node_modules/carbon-components-react/es/components/Accordion/Accordion.js:58:30
bo
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:197349
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:214199
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:237041
t
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:213999
evaluate
https://uymgo.csb.app/node_modules/carbon-components-react/es/components/Accordion/index.js:4:23
bo
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:197349
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:214199
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:237041
t
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:213999
evaluate
https://uymgo.csb.app/node_modules/carbon-components-react/es/index.js:538:23
bo
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:197349
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:214199
value
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:237041
t
https://codesandbox.io/static/js/sandbox.cc47295ac.js:1:213999
evaluate
/src/index.js:3
```

I assume that is not expected!

**Comment by emyarod** [comment]
it appears our code sandboxes have broken in the last few days, not sure as to why at the moment

I have raised a discussion with the Code Sandbox team to figure out what has gone wrong as it seems the package is being built differently

**Comment by matthew-chirgwin** [comment]
Thanks for confirming @emyarod - can you let me know when it is resolved? I will then write a recreate for this issue.

**Comment by emyarod** [comment]
sure, will keep you posted @matthew-chirgwin

**Comment by emyarod** [comment]
@matthew-chirgwin the code sandbox template issue should now be resolved!

**Comment by matthew-chirgwin** [comment]
Thanks for the update @emyarod ! I have created the sceanrio here:

https://codesandbox.io/s/admiring-beaver-n1gzs?file=/src/index.js

So far - this is working as expected (yay!) - ie focus is staying on the copy button. I will dig into it further and see if there is anything we are doing different vs the sandbox/update it as needed

**Comment by dakahn** [comment]
Closing as resolved, but will reopen if need be
